### PR TITLE
Spec

### PR DESCRIPTION
**Pull request by manandearth**
## Goal
Have the app type-resilient and test-proof 
## Proposed changes
- Add tests for every aspect
- Get all possible specs
- Catch all exceptions

## PR status, roadblocks, etc
1. Not every stage is spec'd
2. How does spec helps dealing with the wrong output (exceptions)
 
## PR checklist 

• [ ] any bugs?                                               -> still no wrong input is catched 
• [ ] any duplication?                                      -> Is it ok to spec weight and height seperately, it looks fine to me.
• [x] any leftover or commented-out code?
• [x] any bad formatting?
• [-] is MVC followed?

### CODE REVIEW

**Comment by manandearth** [comment]
Hi @vemv . I addressed the issues. look above at my doubts regarding spec and how..
Feel free to just ask me to keep working and review once I have more tests. I submitted this just as it is answering the previous review.

File: src/bmi_calculator/core.clj [hunk 1]
```diff
@@ -5,11 +5,11 @@
             [orchestra.spec.test :as st])
   (:gen-class))
 
-(spec/def ::weight (spec/and double? (complement empty?)))
+(spec/def ::weight double?)
 (spec/def ::height double?)
 (spec/def ::bmi    (spec/and pos? double?))
 (spec/def ::present-string (spec/and string? (complement empty?)))
-(spec/def ::values-map map?)
+(spec/def ::values-map (spec/keys :req-un [::weight ::height]))
```
Comment (vemv) [inline src/bmi_calculator/core.clj]: `values-map` is not an optimal name

Why? b/c when I find it in code, I'll have no idea what it means

Better: `::person`

File: src/bmi_calculator/core.clj [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
-               (<= num 0)  "not likely"
-               (<= num 16) "dangerously underweight"
-               (<= num 20) "underweight"
-               (<= num 25) "normal"
-               (<= num 30) "overweight.."
-               (<= num 40) "obese"
-               (<= num 60) "extremely obese"
-               :else      "not possible"))
+(defn-spec statement ::present-string
+  [num ::bmi] (cond
```
Comment (vemv) [inline src/bmi_calculator/core.clj]: Always place a newline after the args vector

File: test/bmi_calculator/core_test.clj [hunk 3]
```diff
@@ -10,28 +10,15 @@
       1.0 {:x 100.0 :y 10.0}
       )))
 
-(defn get-input-helper
```
Comment (vemv) [inline test/bmi_calculator/core_test.clj:13]: Get rid of destructuring in `bmi-test` too

Tip: whenever receving feedback, search for more ocurrences of that issue

**Review by vemv — APPROVED** [review]

**Comment by vemv** [comment]
Great work and happy you're on board with `defn-spec`!

**Pull request merged**